### PR TITLE
Update data.ipynb

### PR DESCRIPTION
**Pull request by rastala**
# Description

Describe user identity based authentication for data

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/assets/data/data.ipynb [hunk 1]
```diff
@@ -221,6 +221,33 @@
...
+   "source": [
+    "from azure.ml.Identity import AzureMLOnBehalfOfCredential\r\n",
+    "\r\n",
+    "job = command(\r\n",
+    "    code=\"./src\",\r\n",
+    "    command=\"python read_data.py --input_data ${{inputs.input_data}}\",\r\n",
+    "    inputs=my_job_inputs,\r\n",
+    "    environment=\"AzureML-sklearn-1.0-ubuntu20.04-py38-cpu:1\",\r\n",
+    "    compute=\"cpu-cluster\",\r\n",
+    "    identity= AzureMLOnBehalfOfCredential() \r\n",
```
Comment (ganzhi) [inline sdk/python/assets/data/data.ipynb:247]: Think we should use UserIdentity object here:  https://learn.microsoft.com/en-us/python/api/azure-ai-ml/azure.ai.ml.useridentity?view=azure-python-preview

Comment (rastala) [reply sdk/python/assets/data/data.ipynb]: Fixed

**Review by ganzhi — APPROVED** [review]

File: sdk/python/assets/data/data.ipynb [hunk 2]
```diff
@@ -221,6 +221,33 @@
     "returned_job.studio_url"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Authenticate with user identity\r\n",
+    "\r\n",
+    "When running a job on a compute cluster, you can also use your user identity to access data. To enable job to access data on behald of you, specify **identity= AzureMLOnBehalfOfCredential()** in job definition, as shown below. For more details, see [Accessing storage services](https://learn.microsoft.com/azure/machine-learning/how-to-identity-based-service-authentication)"
```
Comment (ChunyuMSFT) [inline sdk/python/assets/data/data.ipynb:230]: AzureMLOnBehalfOfCredential here should be UserIdentity

Comment (rastala) [reply sdk/python/assets/data/data.ipynb]: Fixed

**Review by ChunyuMSFT — APPROVED** [review]

File: sdk/python/assets/data/data.ipynb [hunk 3]
```diff
@@ -339,22 +366,16 @@
...
    "description": "Read, write and register a data asset"
   },
+  "interpreter": {
+   "hash": "125dd140da12d11a54b5332fbf8107b6ee18903a1d28720c7f05fc0e3013642d"
+  },
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
-   "language": "python",
-   "name": "python310-sdkv2"
+   "display_name": "Python 3.7.6 64-bit (conda)",
```
Comment (harneetvirk) [inline sdk/python/assets/data/data.ipynb:373]: Please don't override the kernel spec.

Comment (rastala) [reply sdk/python/assets/data/data.ipynb]: Fixed

**Review by harneetvirk — APPROVED** [review]

**Pull request merged**